### PR TITLE
Feature/015 create product screen

### PR DESCRIPTION
**Pull request by halfangel37**
Feature:Create product screen
New Update Demo :
https://www.loom.com/share/c470f35fb97a4447b2d48c3603bfc7aa
Demo Update:
Add Image Url field and validate currency duplicate:
https://www.loom.com/share/407e8b7634c74448a80e35154c6c317f

### CODE REVIEW

File: src/product/views/CreateProduct.vue [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  </v-main>
+</template>
+
+<script>
+import CreateProductForm from "../components/CreateProductForm.vue";
+export default {
+  components: { CreateProductForm },
+  data: () => ({
+    isButtonDisabled: false,
+  }),
```
Comment (truonghungit) [inline src/product/views/CreateProduct.vue:23]: `isButtonDisabled` is not clear, rename to `isPending` that means we are waiting for the request to the server

File: src/product/components/CreateProductForm.vue [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  props: {
+    isButtonDisabled: Boolean,
+  },
+  data: () => ({
+    nameRules: [validators.required("Product name is required")],
+    typeRules: [validators.required("Product type is required")],
+    priceRules: [validators.required("Product price is required")],
+    currencyRules: [validators.required("Currency is required")],
+    type: ["Item", "Service"],
+    switchLabel: ["Enabled", "Disabled"],
```
Comment (truonghungit) [inline src/product/components/CreateProductForm.vue:83]:
```suggestion
```
Remove if we don't use

File: src/product/components/CreateProductForm.vue [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+export default {
+  props: {
+    isButtonDisabled: Boolean,
+  },
+  data: () => ({
+    nameRules: [validators.required("Product name is required")],
+    typeRules: [validators.required("Product type is required")],
+    priceRules: [validators.required("Product price is required")],
+    currencyRules: [validators.required("Currency is required")],
+    type: ["Item", "Service"],
```
Comment (truonghungit) [inline src/product/components/CreateProductForm.vue:82]:
```suggestion
    productTypes: ["Item", "Service"],
```

**Review by truonghungit — APPROVED** [review]
All good, just small comments, please resolve before merge into the main

File: src/product/components/CreateProductForm.vue [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    />
+    <v-select
+      v-model="product.productPrices[0].currency"
+      outlined
+      dense
+      :items="currencies"
+      label="Currency"
+      :rules="currencyRules"
+      validate-on-blur
+    ></v-select>
```
Comment (luantc13) [inline src/product/components/CreateProductForm.vue:45]: `product_prices` is a list, with many pairs of `currency - price`, pls re-implement this form.

**Review by luantc13 — CHANGES_REQUESTED** [review]
Nice work!
However, there are a big issue - `product_prices` here. Pls resolve ít and we can merge after that.

**Review by luantc13 — APPROVED** [review]
Nice work!

**Pull request merged**